packages/http/src/router/matchPath.ts

```diff
@@ -21,10 +27,20 @@ export function matchPath(requestPath: string, operationPath: string): E.Either<
     return E.right(MatchType.NOMATCH);
   }
 
+  // a/b:c = a/b:c
+  const operationPathColonFragments = fragmentarizeWithColon(operationPath);
+  const requestPathColonFragments = fragmentarizeWithColon(requestPath);
+  if (
```

Comment:
Such a check doesn't check the order in separators. 
```
  test('it does not match if separators are not equal', () => {
    assertRight(matchPath('a:b/c', 'a/b:c'), e => expect(e).toEqual(MatchType.NOMATCH));
  });
```

I'd try to simplify this method. 2 kinds of separations a bit confusing. Considering that at the end we will use only `fragmentarizeWithColon`.  We could check if separators are equal in the operation and request.

```
import { MatchType } from './types';
import * as E from 'fp-ts/Either';
import { isEqual } from 'lodash';

const pathSeparatorsRegex = /[/:]/g;

function fragmentarize(path: string): string[] {
  return path.split(pathSeparatorsRegex).slice(1);
}

function getTemplateParamName(pathFragment: string) {
  const match = /{(.*)}/.exec(pathFragment);
  return match && match[1];
}

function isSeparationEqual(path1: string, path2: string): boolean {
  return isEqual(path1.match(pathSeparatorsRegex), path2.match(pathSeparatorsRegex));
}

export function matchPath(requestPath: string, operationPath: string): E.Either<Error, MatchType> {
  if (!isSeparationEqual(requestPath, operationPath)) {
    return E.right(MatchType.NOMATCH);
  }

  const operationPathFragments = fragmentarize(operationPath);
  const requestPathFragments = fragmentarize(requestPath);
  if (operationPathFragments.length !== requestPathFragments.length) {
    return E.right(MatchType.NOMATCH);
  }

  const params = [];
  while (requestPathFragments.length) {
    const requestPathFragment = requestPathFragments.shift();
    const operationPathFragment = operationPathFragments.shift();

    const paramName = getTemplateParamName(operationPathFragment as string);

    if (paramName === null && operationPathFragment !== requestPathFragment) {
      // if concrete fragment and fragments are different return false
      return E.right(MatchType.NOMATCH);
    } else if (paramName !== null) {
      params.push({
        name: paramName,
        value: requestPathFragment,
      });
    }
  }

  return E.right(params.length ? MatchType.TEMPLATED : MatchType.CONCRETE);
}

```